### ISSUE TITLE
values does not work on Slack App approach

### ISSUE THREAD

**Issue by DSdatsme** [slackapi/slack-github-action#171]
### Description

The GitHub actions are giving an error for the `values` field when being used. I've copied the sample code from here
https://github.com/slackapi/slack-github-action/blob/main/example-workflows/Technique_2_Slack_App/JSON_payload.yml#L45-L47

I am using the OAuth approach (approach 2) of the slack bot to send notifications to slack.
Need to understand if I am missing something or if it's for another setup? because documentation does not mention anything about `values` field.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [ ] enhancement (feature request)
- [x] question
- [ ] documentation related
- [x] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have made my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

#### Reproducible in: 

package version:
tested on
- slackapi/slack-github-action@v1.16.0
- slackapi/slack-github-action@v1.23.0

node version:

OS version(s): ubuntu-22.04 (github hosted runners)

#### Steps to reproduce:

1. Copy the example code and run it

#### Expected result:

values field should work
#### Actual result:
getting error

#### Attachments:
<img width="1017" alt="Screenshot 2023-01-24 at 2 36 55 PM" src="https://user-images.githubusercontent.com/23118559/214251914-1ffcd44f-2ed1-41aa-87ee-d834442b8e13.png">

Logs, screenshots, screencast, sample project, funny gif, etc.

**Comment by mwbrooks** [comment]
Hi @DSdatsme 👋🏻 Thanks for reaching out!

I don't see an immediately problem after looking over your code sample and I see you've already tried `slackapi/slack-github-action@v1.23.0`.

I'm going to setup an example to recreate your problem and I'll get back to you shortly! Thanks for your patience 🙇🏻

**Comment by mwbrooks** [comment]
Hi @DSdatsme 👋🏻 🌙 

I've setup an [example repository](https://github.com/mwbrooks/slack-github-action-example/tree/main/.github/workflows) that uses the Slack GitHub Action in 2 ways:

1. [Send a message as a string](https://github.com/mwbrooks/slack-github-action-example/blob/main/.github/workflows/slack-message.yml)
2. [Send a message as JSON payload](https://github.com/mwbrooks/slack-github-action-example/blob/main/.github/workflows/slack-message-json.yml)

I can confirm that I was able to re-create your error with JSON Payload example. 

I solved the problem by removing the `values: |` property and instead referencing the values using the [GitHub Context syntax](https://docs.github.com/en/actions/learn-github-actions/contexts) (`${{ <context> }}`):

[Example: .github/workflows/slack-message-json.yml#L34-L36](https://github.com/mwbrooks/slack-github-action-example/blob/main/.github/workflows/slack-message-json.yml#L34-L36):
```
  "text": "${{ github.sha }}"
},
"url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

I believe the examples are slightly out-of-date. They should not use the `values:` property and instead they should use the new `${{ github }}` context syntax. I'll try to update those examples for us.

I'd recommend that you look at the examples in the [README.md for Technique 2](https://github.com/slackapi/slack-github-action#technique-2-slack-app). They appear to be accurate and should unblock you.

Let me know how it goes!

**Comment by DSdatsme** [comment]
Heya @mwbrooks ,
Thanks for the quick response on the issue.

My goal for using `values` block was to pass on GitHub workflow inputs (workflow parameters) to slack message.
something like `{{ inputs.UserInput1 }}`.
Currently, this action only supports `github` context, I thought `values` would be a good way to pass on other variables which are outside of github context.

**Comment by DSdatsme** [comment]
Or maybe I'll have to wait for `env` context implementation(PR 159) which looks like blocked due to test cases.
As a temporary workaround, I've forked this repo and implemented the env context and started using it for my actions.

Will have to wait until either values or env context feature is released, and any other better workaround you could suggest.

**Comment by mwbrooks** [comment]
Hey @DSdatsme, thanks for the response and background on why you wanted to use `values`.

Please checkout the [Contexts documentation](https://docs.github.com/en/actions/learn-github-actions/contexts) because GitHub Workflow inputs are also available: `${{ inputs }} and `${{ inputs.<name> }}`

Let me know if that helps! It would be ideal if you can use this repo instead of maintaining your own fork 😄

**Comment by DSdatsme** [comment]
Hey @mwbrooks ,
Currently, this slack action does not support `inputs` context.
Also, a lot of `github` context variables do not work when I used them in payload json. I am unsure whether it's an issue with reading the file or I am missing out something.

I've created a PR to your example repo to explain this with an example.
`github.actor` works whereas `github.repository` or `github.run_id` does not seem to work.

This is the payload template that I used

```json
{
            "type": "section",
            "text": {
                "type": "mrkdwn",
                "text": "URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }} \n GH actor: ${{ github.actor }},,,,,, GH Input: ${{ inputs.branch }}"
            }
        }
```

This is what I got as a slack message:
<img width="448" alt="Screenshot 2023-01-30 at 11 53 52 AM" src="https://user-images.githubusercontent.com/23118559/215403318-39a5a805-7ba3-4b86-8a93-709e25506af5.png">

Unsure if you want to track it as a separate 2 issues or continue here.
1) not all `github` context variables are working.
2) support for `inputs` and other contexts.

**Comment by zimeg** [comment]
@DSdatsme the issue around missing `github` context variables for `payload-file-path` are being tracked in #203!

**Comment by DSdatsme** [comment]
Hey @zimeg , yes it's missing some context variables.
Thanks for mentioning the issue, I see other folks have also faced the same issue.

As I mentioned above, as a workaround, I am using this patched version for myself.
https://github.com/slackapi/slack-github-action/compare/main...DSdatsme:slack-github-action:main

**Comment by zimeg** [comment]
👋 Hello again! Refreshing myself on this issue, I noticed that #159 was moved to #200 and was included in the [`v1.24.0`](https://github.com/slackapi/slack-github-action/releases/tag/v1.24.0) release!

It's super neat to see you've been supporting this fix in the meantime! ❤️ 

I'll close this issue now that support is available in both places, and also wanted to share that [`@v2.0.0`](https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0) was released and continues to support this, albeit with multiple changes to inputs 🔍

Please feel free to follow up with additional ideas on inputs or other questions! And to share a few more resources about expected inputs...

> 1. not all github context variables are working.

This is unfortunately [expected](https://github.com/slackapi/slack-github-action/blob/e598089eaef53883a2d9b325e044899548518a03/src/content.js#L160-L163) when using `payload-file-path` with `payload-templated` set to `true`. We've improved documentation for this case, and the expected values are available as [the default GitHub event](https://github.com/slackapi/slack-github-action?tab=readme-ov-file#sending-variables) contexts.

> 2. support for inputs and other contexts.

I'm interested in hearing more about this! Please feel free to [open a new issue](https://github.com/slackapi/slack-github-action/issues/new) if you're still finding the current features aren't supporting your workflows. I'm not so familiar with `values` without [references](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions) and am hoping `env` might be sufficient, but please let me know 🙏
